src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemon.java

```diff
@@ -140,11 +143,26 @@ public static void main(String[] args) {
         ExecutorService executorService = config.getExecutorService();
 
         // Daemon
-        MultiLangDaemon daemon = new MultiLangDaemon(
+        final MultiLangDaemon daemon = new MultiLangDaemon(
                 config.getKinesisClientLibConfiguration(),
                 config.getRecordProcessorFactory(),
                 executorService);
 
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            @Override
+            public void run()
+            {
+                LOG.info("Process terminanted, will initiate shutdown.");
+                try {
+                    Future<Void> fut = daemon.worker.requestShutdown();
```

Comment:
The way dispatch is handled in the worker already guarantees this.  The record processors are represented as a state machine, and all request shutdown does is request that on the change event to go to the Shutdown Notification Requested state.  [This is the diagram for the state changes](https://github.com/awslabs/amazon-kinesis-client/blob/master/src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ConsumerStates.java#L3).